### DISCUSSION TITLE
using React Icons instead of Lucide React

### DISCUSSION THREAD

**Discussion by rawand-faraidun** [shadcn-ui/ui#2603] [Ideas]
I believe using `react-icons` is better instead of `lucide-react`, it contains all Lucide Icons and many elses which can be used, I have seen some project that manually replaces the icons library because they also used other icons, and `react-icons` contains most of the popular libraries, and I always end-up replacing too, so I believe it would be better to use it, or having a configuration option to make the decision is awesome.

**Comment by Evavic44** [comment]
I agree. Although I'm more keen to know the performance difference between the two.

**Comment by snoot-booper** [comment]
I disagree. `react-icons` entices you to mix icon sets which don't all match in style.

**Reply by azizoid** [reply]
lucide-react is limited, and depricating brand logos for example. instead of installing a lot of icon libraries, i would feel comfortable having only one. 
also lucide icons do not prefix or suffix their Icon names, so you can import Link from lucide icons, and try to debug the error.

**Reply by calebfrieze** [reply]
That could be a desired effect in some situations. `lucide-react` doesn't have a lot of brand icons for example. it should be left up to the implementer to decide which icons are pulled in. `react-icons` would make it nice to just have one lib

**Reply by rawand-faraidun** [reply]
There are icon sets that match within `react-icons`, having the option to decide which icon library you want is great, maybe for different-looking icons to

In the new `components.json`, there is an option called `iconLibrary`, but I am unsure what the options are for it.

**Reply by Clovel** [reply]
I would also like to know if we wan set the `react-icons` library to `iconLibrary`. The [JSON schema](https://ui.shadcn.com/schema.json) just says "string".

**Reply by codedusting** [reply]
> I would also like to know if we wan set the `react-icons` library to `iconLibrary`. The [JSON schema](https://ui.shadcn.com/schema.json) just says "string".

Found any answer?

**Comment by Prajwal-alphanimble** [comment]
you might wanna check out this discussion where I implemented a new resizable and dragable dashboard layout... this can be very helpfull to many developers in making analytic, sharable dashboard, the users can alter the dashboard based on their liking...

The disscussion link: https://github.com/shadcn-ui/ui/discussions/10466
